crates/rpc-replay/src/transactions.rs

```diff
@@ -38,10 +44,12 @@ fn da_mode_core_to_api(
 }
 
 /// Maps starknet-core transactions to Blockifier-compatible types.
-pub fn starknet_rs_to_blockifier(
+pub async fn starknet_rs_to_blockifier(
     sn_core_tx: &starknet::core::types::Transaction,
+    provider: &JsonRpcClient<HttpTransport>,
+    block_number: u64,
 ) -> Result<blockifier::transaction::transaction_execution::Transaction, Box<dyn Error>> {
```

Comment:
For another PR, but we really need to return something better than `Box<dyn Error>` here.